interpreter/README.md

```diff
@@ -254,13 +254,13 @@ op:
   br_table <var>+
   br_on_null <var>
   br_on_<castop> <var>
-  br_on_cast <var>
+  br_on_cast_canon <var> (type <var>)
   br_on_non_null <var>
   br_on_non_<castop> <var>
-  br_on_cast_fail <var>
+  br_on_cast_canon_fail <var> (type <var>)
```

Comment:
What is `(type <var>)` here? Why isn't it added to other instructions like `struct.new` below?

Reply:
This is the type to cast to, as it can no longer be derived from the RTT operand. Most other instructions already had a (previously redundant) explicit type (cf the discussion regarding type annotations).

The syntax is chosen to mirror other cases like `call_indirect`, and to avoid confusion with the branch label.

I'd be fine with switching to `(type _)` syntax everywhere, it would be more consistent – but also more verbose. I justified the difference to myself by the prevalent choice that indices associated with the primary object kind of an instruction do not have a sort prefix (e.g., the global index in `global.get`, the table index in `call_indirect`, the label index in branches) whereas secondary ones do (e.g., the type index in `call_indirect` and `br_on_cast`, the memory index in `data`). And one can argue that the type in GC instructions represents the primary object kind, if that makes sense. :)